src/commands/env/logdrain/remove.ts

```diff
@@ -75,12 +76,36 @@ export default class LogDrainRemove extends Command {
       this.warn(messages.getMessage('flags.url.deprecation'));
     }
 
-    const appName = await resolveAppNameForEnvironment(targetCompute);
+    try {
+      const appName = await resolveAppNameForEnvironment(targetCompute);
 
-    cli.action.start(`Deleting drain for environment ${herokuColor.app(targetCompute)}`);
+      if (flags.json) {
+        await this.client.delete<Heroku.LogDrain>(`apps/${appName}/log-drains/${encodeURIComponent(url)}`);
```

Comment:
This part shouldn't be wrapped by the json flag, as it applies to both cases. Only wrap the json output.